<a href="https://colab.research.google.com/github/MaureenGatu/Data-Analysis/blob/main/Global_data_on_confirmed_COVID_19_deaths.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***COVID-19 DEATHS DATA EXPLORATION***

## 1.0 Connecting to our Database

In [2]:
# We will first load the sql extension into our environment
%load_ext sql

# Then connect to our in memory sqlite database
# NB: This database will cease to exist as soon as the database connection is closed. 
%sql sqlite://

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


'Connected: @None'

## 1.1 Importing Data from CSV files

Datasets used in this notebook:


1.   **Covid 19 Deaths**
2.   **Covid 19 Vaccinations** 

> The dataset is from Our World in Data. ([Dataset Download](https://ourworldindata.org/covid-deaths))






In [3]:
# Importing the pandas library
# We will use a function read_csv from pandas to read our datasets as shown
#
import pandas as pd 

In [4]:
# Loading our table from the respective CSV files 
with open('/content/drive/MyDrive/Data Science/Data Analysis/SQL/DATA/COVID_DEATHS_DATA.csv','r') as f:
    covid_deaths = pd.read_csv(f, index_col=0, encoding='utf-8')
# saving the urban center CSV file into a database
%sql DROP TABLE if EXISTS covid_deaths;
%sql PERSIST covid_deaths;

# Displaying the first  5 entries 
%sql SELECT * FROM covid_deaths LIMIT 5;

 * sqlite://
Done.
 * sqlite://
 * sqlite://
Done.


[('AFG', 'Asia', 'Afghanistan', '24/02/2020', 39835428.0, 5.0, 5.0, None, None, None, None, 0.126, 0.126, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '25/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '26/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '27/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '28/02/2020', 39835428.0, 5.0, 0.0, None, None, None, None, 0.126, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None)]

In [5]:
# Let's load our covid_vaccine dataset below
with open('/content/drive/MyDrive/Data Science/Data Analysis/SQL/DATA/COVID_VACCINATION_DATA.csv','r') as f:
    covid_vaccine = pd.read_csv(f, index_col=0, encoding='utf-8') 

# Save the houshold dataset into a database
%sql DROP TABLE if EXISTS covid_vaccine;
%sql PERSIST covid_vaccine;

# Display the first 5 entries from the database
%sql SELECT * FROM covid_vaccine LIMIT 5;

 * sqlite://
Done.
 * sqlite://
 * sqlite://
Done.


[('AFG', 'Asia', 'Afghanistan', '24/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '25/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '26/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '27/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None),
 ('AFG', 'Asia', 'Afghanistan', '28/02/2020', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 54.422, 18.6, 2.5810000000000004, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None, None, None, None)]

In [6]:
#Selecting the data that we are going to be using ordered by date and location
%%sql

SELECT location, date, total_cases,new_cases, total_deaths, population FROM covid_deaths ORDER BY 1,2 LIMIT 10;

 * sqlite://
Done.


[('Afghanistan', '01/01/2021', 52513.0, 183.0, 2201.0, 39835428.0),
 ('Afghanistan', '01/02/2021', 55059.0, 36.0, 2404.0, 39835428.0),
 ('Afghanistan', '01/03/2020', 5.0, 0.0, None, 39835428.0),
 ('Afghanistan', '01/03/2021', 55733.0, 19.0, 2444.0, 39835428.0),
 ('Afghanistan', '01/04/2020', 192.0, 26.0, 4.0, 39835428.0),
 ('Afghanistan', '01/04/2021', 56517.0, 63.0, 2489.0, 39835428.0),
 ('Afghanistan', '01/05/2020', 2171.0, 344.0, 64.0, 39835428.0),
 ('Afghanistan', '01/05/2021', 59939.0, 194.0, 2631.0, 39835428.0),
 ('Afghanistan', '01/06/2020', 15836.0, 656.0, 269.0, 39835428.0),
 ('Afghanistan', '01/06/2021', 72977.0, 1139.0, 2973.0, 39835428.0)]

In [76]:
%%sql
--Looking at total deaths vs total cases
SELECT location, date, total_cases, total_deaths, (total_deaths/total_cases)*100 as death_percentage
FROM covid_deaths WHERE total_cases AND total_deaths IS NOT NULL
ORDER BY 5 desc LIMIT 10;

 * sqlite://
Done.


[('Guyana', '12/03/2020', 1.0, 1.0, 100.0),
 ('Guyana', '13/03/2020', 1.0, 1.0, 100.0),
 ('Guyana', '14/03/2020', 1.0, 1.0, 100.0),
 ('Iran', '19/02/2020', 2.0, 2.0, 100.0),
 ('Peru', '06/03/2020', 1.0, 1.0, 100.0),
 ('Peru', '07/03/2020', 1.0, 1.0, 100.0),
 ('San Marino', '29/02/2020', 1.0, 1.0, 100.0),
 ('San Marino', '01/03/2020', 1.0, 1.0, 100.0),
 ('San Marino', '02/03/2020', 1.0, 1.0, 100.0),
 ('Sudan', '14/03/2020', 1.0, 1.0, 100.0)]

In [53]:
%%sql
--Looking at total deaths vs total cases
SELECT location, date, total_cases, total_deaths, (total_deaths/total_cases)*100 as death_percentage
FROM covid_deaths 
WHERE location like '%Kenya%' AND  total_cases AND total_deaths IS NOT NULL 
ORDER BY 5 desc LIMIT 5;

 * sqlite://
Done.


[('Kenya', '16/05/2020', 830.0, 50.0, 6.024096385542169),
 ('Kenya', '15/05/2020', 781.0, 45.0, 5.761843790012804),
 ('Kenya', '17/05/2020', 887.0, 50.0, 5.636978579481398),
 ('Kenya', '14/05/2020', 758.0, 42.0, 5.540897097625329),
 ('Kenya', '18/05/2020', 912.0, 50.0, 5.482456140350877)]

In [67]:
%%sql
--Looking at percentage of infection per country 
  --Ordered by highest infected population percentage
SELECT location, date, total_cases, total_deaths, (total_cases/population)*100 as infected_pop_percentage
FROM covid_deaths 
GROUP BY location
ORDER BY 5 desc LIMIT 5;

 * sqlite://
Done.


[('Seychelles', '08/10/2021', 21726.0, 118.0, 21.965423111919925),
 ('Montenegro', '08/10/2021', 134602.0, 1980.0, 21.43169901807337),
 ('Andorra', '08/10/2021', 15291.0, 130.0, 19.76756211702045),
 ('San Marino', '08/10/2021', 5466.0, 91.0, 16.07174360482211),
 ('Czechia', '08/10/2021', 1698061.0, 30496.0, 15.833396506129441)]

Seychelles, Montenegro and Andorra had the highest percentage of infeccted population which were 21.97%, 21.43% and 19.77% respectively as of 08/10/2021.

In [97]:
%%sql
--Looking at top percentage of infected population in Kenya.
SELECT location, total_cases, MAX(total_deaths) as highest_total_deaths, MAX((total_cases/population))*100 as infected_pop_percentage
FROM covid_deaths
WHERE continent IS NOT NULL
GROUP BY location  
ORDER BY 4 desc LIMIT 15;

 * sqlite://
Done.


[('Seychelles', 21726.0, 118.0, 21.965423111919925),
 ('Montenegro', 134602.0, 1980.0, 21.43169901807337),
 ('Andorra', 15291.0, 130.0, 19.76756211702045),
 ('San Marino', 5466.0, 91.0, 16.07174360482211),
 ('Czechia', 1698061.0, 30496.0, 15.833396506129441),
 ('Georgia', 628719.0, 9192.0, 15.797860832766089),
 ('Bahrain', 275538.0, 1390.0, 15.760383688107558),
 ('Maldives', 85394.0, 233.0, 15.708399249475738),
 ('Israel', 1302777.0, 7885.0, 14.821503983719268),
 ('Slovenia', 300084.0, 4899.0, 14.435978242411327),
 ('Serbia', 994708.0, 8631.0, 14.398896156233498),
 ('Cyprus', 120992.0, 558.0, 13.503496074240656),
 ('United States', 44290052.0, 712693.0, 13.303708801122054),
 ('Lithuania', 347548.0, 5179.0, 12.920662844413581),
 ('Luxembourg', 79018.0, 836.0, 12.447425545120302)]

Percentage of Kenyan population infected with covid 19 as at 08/10/2021 was 0.456%

In [103]:
%%sql
--Looking at coutries with the highest total death count.
SELECT location, total_cases, MAX(total_deaths) as maximum_deaths 
FROM covid_deaths 
WHERE continent IS NOT NULL
group by location
ORDER BY 3 desc LIMIT 5;

 * sqlite://
Done.


[('United States', 44290052.0, 712693.0),
 ('Brazil', 21550730.0, 600425.0),
 ('India', 33935309.0, 450375.0),
 ('Mexico', 3714392.0, 281610.0),
 ('Russia', 7602386.0, 210673.0)]

In [99]:
%%sql
-- Top 5 countries with highest death count per population
SELECT location, MAX((total_deaths/population))*100 as death_count_per_population
FROM covid_deaths
-- To exclude data that was aggregated per continent
WHERE continent IS NOT NULL
--Group by location to get highest per country
GROUP BY location 
--Order in descending order of the highest death count per location 
ORDER BY 2 desc LIMIT 5;


 * sqlite://
Done.


[('Peru', 0.5984277601990323),
 ('Bosnia and Herzegovina', 0.332867672000782),
 ('North Macedonia', 0.3264573543173853),
 ('Montenegro', 0.31526102179600063),
 ('Hungary', 0.3142463246933153)]

Peru had the highest death to population ratio at 0.60% percent followed by Bosmia and Herzegovina at 0.33%.

In [ ]:
%%sql


In [ ]:
%%sql


In [ ]:
%%sql


In [ ]:
%%sql


In [ ]:
%%sql


In [ ]:
%%sql


In [ ]:
%%sql


In [ ]:
%%sql


In [ ]:
%%sql


In [ ]:
%%sql


In [ ]:
%%sql


In [ ]:
%%sql
